tests/infra/jsonrpc.py

```diff
@@ -315,6 +318,126 @@ def rpc(self, method, params):
         return self.response(id)
 
 
+class CurlClient:
+    def __init__(
+        self,
+        host,
+        port,
+        server_hostname,
+        cert,
+        key,
+        cafile,
+        version,
+        format,
+        description,
+    ):
+        self.host = host
+        self.port = port
+        self.server_hostname = server_hostname
+        self.cert = cert
+        self.key = key
+        self.cafile = cafile
+        self.version = version
+        self.format = format
+        self.stream = Stream(version, format=format)
+        self.pending = {}
+
+    def signed_request(self, method, params):
+        r = self.stream.request(method, params)
+        with tempfile.NamedTemporaryFile() as nf:
+            msg = getattr(r, "to_{}".format(self.format))()
+            LOG.debug("Going to send {}".format(msg))
+            nf.write(msg)
+            nf.flush()
+            dgst = subprocess.run(
+                ["openssl", "dgst", "-sha256", "-sign", "member1_privk.pem", nf.name],
+                check=True,
+                capture_output=True,
+            )
+            subprocess.run(["cat", nf.name], check=True)
+            cmd = [
+                "curl",
+                "-v",
+                "-k",
+                f"https://{self.server_hostname}:{self.port}/",
+                "-H",
+                "Content-Type: application/json",
+                "-H",
+                f"Authorize: {base64.b64encode(dgst.stdout).decode()}",
+                "--resolve",
+                f"{self.server_hostname}:{self.port}:{self.host}",
+                "--data-binary",
+                f"@{nf.name}",
+            ]
+            if self.cafile:
+                cmd.extend(["--cacert", self.cafile])
+            if self.key:
+                cmd.extend(["--key", self.key])
+            if self.cert:
+                cmd.extend(["--cert", self.cert])
+            LOG.debug(f"Running: {' '.join(cmd)}")
+            rc = subprocess.run(cmd, capture_output=True)
+            LOG.debug(f"Received {rc.stdout.decode()}")
+            if rc.returncode != 0:
+                LOG.debug(f"ERR {rc.stderr.decode()}")
+            self.stream.update(rc.stdout)
+        return r.id
+
+    def request(self, method, params):
+        r = self.stream.request(method, params)
+        with tempfile.NamedTemporaryFile() as nf:
+            msg = getattr(r, "to_{}".format(self.format))()
+            LOG.debug("Going to send {}".format(msg))
+            nf.write(msg)
+            nf.flush()
+            cmd = [
```

Comment:
Worth adding a comment why we use curl CLI instead of requests?

Reply:
Yes, this is only until we split the frontends #237